worker/worker.go

```diff
@@ -0,0 +1,113 @@
+package worker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/bluesky-social/indigo/api/bsky"
+	"github.com/bluesky-social/indigo/xrpc"
+	"github.com/jackc/pgx/v5"
+	"github.com/strideynet/bsky-furry-feed/bluesky"
+	"github.com/strideynet/bsky-furry-feed/store"
+	"go.uber.org/zap"
+)
+
+func Start(
+	ctx context.Context,
+	log *zap.Logger,
+	pdsHost string,
+	bskyCredentials *bluesky.Credentials,
+	pgxStore *store.PGXStore,
+) error {
+	client, err := bluesky.ClientFromCredentials(ctx, pdsHost, bskyCredentials)
+	if err != nil {
+		return err
+	}
+	bgsClient := bluesky.BGSClient{}
+
+	ticker := time.NewTicker(time.Second * 5)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-ticker.C:
+			task, err := pgxStore.GetNextFollowTask(ctx)
+			if errors.Is(err, pgx.ErrNoRows) {
+				continue
+			}
+
+			log.Info("processing task", zap.Int64("id", task.ID))
+
+			if task.ShouldUnfollow {
```

Comment:
Let's move the big if statement into a function - I can see there's a few cases here where an error will result in the handler crashing rather than the task being marked in an error state because it's a little confusing what `return` does in this scope.

Reply:
done in f1dab899cc256f5964e76195865f9114cb840a04